src/library_exceptions.js

```diff
@@ -7,92 +7,196 @@
 var LibraryExceptions = {
   __exception_last: '0',
   __exception_caught: ' []',
-  __exception_infos: '{}',
-
-  __exception_deAdjust__deps: ['__exception_infos'],
-  __exception_deAdjust: function(adjusted) {
-    if (!adjusted || ___exception_infos[adjusted]) return adjusted;
-    for (var key in ___exception_infos) {
-      var ptr = +key; // the iteration key is a string, and if we throw this, it must be an integer as that is what we look for
-      var adj = ___exception_infos[ptr].adjusted;
-      var len = adj.length;
-      for (var i = 0; i < len; i++) {
-        if (adj[i] === adjusted) {
-#if EXCEPTION_DEBUG
-          err('de-adjusted exception ptr ' + adjusted + ' to ' + ptr);
+
+  __ExceptionInfoAttrs: {
+    TYPE_OFFSET: 0,
+    DESTRUCTOR_OFFSET: Runtime.POINTER_SIZE,
+    REFCOUNT_OFFSET: Runtime.POINTER_SIZE * 2,
```

Comment:
Does the refcount take 8 bytes? Based on the next field's offset it seems like it is, but based on the operations on it it seems otherwise.

Reply:
Refcount offset is `Runtime.POINTER_SIZE * 2`, next field offset is `Runtime.POINTER_SIZE * 2 + 4` so it takes 4 bytes. The only question is type field which takes `POINTER_SIZE` I do not remember why, but probably because of alignment (pointers indeed should be aligned on pointer size). I will change fields order, it is common practice to sort them in descending order by size to minimize paddings.